### ISSUE TITLE
[LeakScope] 9 Android lifecycle/memory violations detected

### ISSUE THREAD

**Issue by MuhammadTehamsibAliTashfeen** [TrianguloY/URLCheck#575]
## LeakScope: Android Lifecycle & Memory Leak Violations

> **About this report:** This issue was automatically generated by **LeakScope**, a static analysis tool for Android lifecycle violations and memory leaks built on the [Soot](https://github.com/soot-oss/soot) framework. This is part of an ongoing academic research study targeting ICSE 2027. No immediate action is required — we would greatly appreciate your feedback on whether these findings are accurate.

### Summary

LeakScope detected **9 potential issue(s)** across **2 detector type(s)**:

| Severity | Count |
|---|---|
| 🔴 High | 0 |
| 🟡 Medium | 1 |
| 🟢 Low (improvement opportunity) | 8 |

| Detector | Count | Severity | Description |
|---|---|---|---|
| `ServiceResourceManagementIssueDetector` | 1 | 🟡 Medium | Service may never stop — missing stopSelf() call |
| `ViewBindingOpportunity` | 8 | 🟢 Low | Manual findViewById() calls — ViewBinding migration opportunity |

---

#### 🟡 `ServiceResourceManagementIssueDetector`
*Service may never stop — missing stopSelf() call*

**Finding #1 — `CustomTabs`**

```
âš ï¸� Service may never stop. Consider calling stopSelf() when work is complete.
  Class: com.trianguloy.urlchecker.services.CustomTabs
  Method: onStartCommand
```

#### 🟢 `ViewBindingOpportunity`
*Manual findViewById() calls — ViewBinding migration opportunity*

**Finding #2 — `MainDialog`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.dialogs.MainDialog
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in initializeModule
  â€¢ findViewById in initializeModule
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #3 — `BackupActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.BackupActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #4 — `AutomationActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.AutomationActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #5 — `ModulesActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.ModulesActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in initModule
  â€¢ findViewById in toggleAllDecorations
  â€¢ findViewById in updateMovableButtons
  â€¢ findViewById in updateMovableButtons
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #6 — `JsonEditorActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.JsonEditorActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #7 — `TutorialActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.TutorialActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in configureBrowserButtons
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #8 — `AboutActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.AboutActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

**Finding #9 — `SettingsActivity`**

```
View Binding Migration Opportunity
Class: com.trianguloy.urlchecker.activities.SettingsActivity
Type: Activity
Current Pattern: Manual view lookup
findViewById() Calls:
  â€¢ findViewById in configureBrowserButtons
  â€¢ findViewById in configureLocale
  â€¢ findViewById in configureTheme
  â€¢ findViewById in configureTheme
  â€¢ findViewById in configureTheme
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
  â€¢ findViewById in onCreate
Benefits of View Binding:
- Eliminates boilerplate findViewById() calls
- Compile-time type safety for view references
- Reduced null pointer exceptions
- Cleaner, more maintainable code
Note: This is a code modernization suggestion, not a memory leak
```

---

**How to respond to this issue:**
- If a finding is a **true positive**: consider applying the recommended fix and closing this issue.
- If a finding is a **false positive**: please leave a comment explaining why — your feedback directly improves our research.
- If you have questions: reply here or open a discussion.

*This report was generated by LeakScope as part of the [ICSE 2027 research artifact](https://github.com/). Tool analyzes compiled APKs using Soot static analysis on [URLCheck](https://github.com/TrianguloY/URLCheck).*
